lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -0,0 +1,185 @@
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: This file provides fundamental utilities for libvirt
+# virtual network test in openqa, for example,
+# tests/virt_autotest/libvirt_host_bridge_virtual_network.pm
+# tests/virt_autotest/libvirt_nated_virtual_network.pm
+# tests/virt_autotest/libvirt_routed_virtual_network.pm
+# tests/virt_autotest/libvirt_isolated_virtual_network.pm
+# Maintainer: Leon Guo <xguo@suse.com>
+
+package virt_autotest::virtual_network_utils;
+
+use base Exporter;
+use Exporter;
+
+use utils;
+use strict;
+use warnings;
+use File::Basename;
+use testapi;
+use Data::Dumper;
+use XML::Writer;
+use IO::File;
+use proxymode;
+use version_utils 'is_sle';
+use virt_autotest_base;
+use virt_utils;
+
+our @EXPORT
+  = qw(download_network_cfg prepare_network restore_standalone destroy_standalone restart_libvirtd restore_guests restore_network destroy_vir_network restore_libvirt_default enable_libvirt_log upload_debug_log check_guest_status);
+
+sub download_network_cfg {
+    #Download required libvird virtual network configuration file
+    my $vnet_cfg_name       = shift;
+    my $wait_script         = "180";
+    my $vnet_cfg_url        = data_url("virt_autotest/$vnet_cfg_name");
+    my $download_cfg_script = "curl -s -o ~/$vnet_cfg_name $vnet_cfg_url";
+    script_output($download_cfg_script, $wait_script, type_command => 0, proceed_on_failure => 0);
+}
+
+sub prepare_network {
+    #Confirm the host bridge configuration file
+    my $virt_host_bridge = shift;
+    my $config_path      = "/etc/sysconfig/network/ifcfg-$virt_host_bridge";
+    if (script_run("[[ -f $config_path ]]") != 0) {
+        assert_script_run("ip link add name $virt_host_bridge type bridge");
+        assert_script_run("ip link set dev $virt_host_bridge up");
+        my $bridge_setup = <<'EOF';
+#!/bin/bash
+. /etc/os-release
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='none'\nSTARTMODE='auto'"
+bridge="BOOTPROTO='dhcp'\nBRIDGE='yes'\nBRIDGE_FORWARDDELAY='0'\nBRIDGE_PORTS='$ACTIVE_NET'\nBRIDGE_STP='off'\nSTARTMODE='auto'"
+> /etc/sysconfig/network/ifcfg-br0.new
+echo -e $bridge >/etc/sysconfig/network/ifcfg-br0
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-br0
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+if [[ $VERSION_ID =~ '11' ]]; then
+    service libvirtd restart;
+else
+    systemctl restart network.service;
+fi
+EOF
+        script_output($bridge_setup, 900);
```

Comment:
Why are you creating bash script? Just use the testapi:
```perl
assert_script_run '. /etc/os-release';
$VERSION_ID = script_output 'echo $VERSION_ID';

$active_net = script_output "ip a|awk -F': ' '/state UP/ {print $2}'|head -n1";

$interface = "BOOTPROTO='none'\nSTARTMODE='auto'";
$bridge = "BOOTPROTO='dhcp'\nBRIDGE='yes'\nBRIDGE_FORWARDDELAY='0'\nBRIDGE_PORTS='$ACTIVE_NET'\nBRIDGE_STP='off'\nSTARTMODE='auto'";

assert_script_run "echo -e \"$bridge\" >/etc/sysconfig/network/ifcfg-br0";
assert_script_run "echo -e \"$interface\" >/etc/sysconfig/network/ifcfg-$active_net";

assert_script_run "cat /etc/sysconfig/network/ifcfg-br0";
assert_script_run "cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET";
if ($VERSION_ID =~ '11') {
    assert_script_run 'service libvirtd restart';
} else {
    systemctl 'restart network.service';
}
```

Reply:
Had updated code changed